snaps/learn/tutorials/transaction-insights.md

```diff
@@ -32,70 +38,493 @@ starter kit by running:
+    width: 100%;
+  }
+`;
+
+const ButtonText = styled.span`
+  margin-left: 1rem;
+`;
+
+const ConnectedContainer = styled.div`
+  display: flex;
+  align-self: flex-start;
+  align-items: center;
+  justify-content: center;
+  font-size: ${(props) => props.theme.fontSizes.small};
+  border-radius: ${(props) => props.theme.radii.button};
+  border: 1px solid ${(props) => props.theme.colors.background?.inverse};
+  background-color: ${(props) => props.theme.colors.background?.inverse};
+  color: ${(props) => props.theme.colors.text?.inverse};
+  font-weight: bold;
+  padding: 1.2rem;
+`;
+
+const ConnectedIndicator = styled.div`
+  content: ' ';
+  width: 10px;
+  height: 10px;
+  border-radius: 50%;
+  background-color: green;
+`;
+
+export const InstallFlaskButton = () => (
+  <Link href="https://metamask.io/flask/" target="_blank">
+    <FlaskFox />
+    <ButtonText>Install MetaMask Flask</ButtonText>
+  </Link>
+);
+
+export const ConnectButton = (props: ComponentProps<typeof Button>) => {
+  return (
+    <Button {...props}>
+      <FlaskFox />
+      <ButtonText>Connect</ButtonText>
+    </Button>
+  );
+};
+
+export const ReconnectButton = (props: ComponentProps<typeof Button>) => {
+  return (
+    <Button {...props}>
+      <FlaskFox />
+      <ButtonText>Reconnect</ButtonText>
+    </Button>
+  );
+};
+
+export const SendHelloButton = (props: ComponentProps<typeof Button>) => {
+  return <Button {...props}>Inoperable</Button>;
+};
+
+export const HeaderButtons = () => {
+  const requestSnap = useRequestSnap();
+  const { isFlask, installedSnap } = useMetaMask();
+
+  if (!isFlask && !installedSnap) {
+    return <InstallFlaskButton />;
+  }
+
+  if (!installedSnap) {
+    return <ConnectButton onClick={requestSnap} />;
+  }
+
+  if (shouldDisplayReconnectButton(installedSnap)) {
+    return <ReconnectButton onClick={requestSnap} />;
+  }
+
+  return (
+    <ConnectedContainer>
+      <ConnectedIndicator />
+      <ButtonText>Connected</ButtonText>
+    </ConnectedContainer>
+  );
+};
+
+````
+</TabItem>
+</Tabs>
+
+
+These three updates are the minimum required to ensure that each user interaction with your Snap is well informed, however, your Snap will function without these tweaks.
+
+### 3. Enable transaction insights and the Ethereum provider
 
 To enable your Snap to provide [transaction insights](../../features/transaction-insights.md) and
 use the global Ethereum provider, request the
 [`endowment:transaction-insight`](../../reference/permissions.md#endowmenttransaction-insight) and
 [`endowment:ethereum-provider`](../../reference/permissions.md#endowmentethereum-provider)
 permissions in `packages/snap/snap.manifest.json`:
 
+<Tabs>
+  <TabItem value="Line updates" label="Update snap.manifest.json" default>
+
 ```json title="snap.manifest.json"
 "initialPermissions": {
   "endowment:transaction-insight": {},
   "endowment:ethereum-provider": {}
 }
 ```
+</TabItem>
+<TabItem value="Full page" label="Example" default>
```

Comment:
Similar to previous comments, I prefer omitting the full file tab.

Reply:
applied